src/main/java/seedu/address/model/client/Client.java

```diff
@@ -110,6 +110,17 @@ public void updateBalance(double amount) {
         balance += amount;
     }
 
+    /**
+     * Checks if the transaction causes a double overflow.
+     *
+     * @param amount The amount of the transaction to be added / deleted.
+     * @return whether the balance would overflow.
+     */
+    public boolean checkIsOverflow(double amount) {
+        double updatedBalance = balance + amount;
```

Comment:
Should we first check if the `amount` overflows? If `amount` is `INFINITY` it is not safe to add it with balance

Reply:
but adding infinity with anything is still infinity i think?

Reply:
`positiveInfinity` + `negativeInfinity` is `NaN`
I know the balance is not infinity but just for safety?

Reply:
ok I added the check